process-manager.js

```diff
@@ -126,13 +126,7 @@ class ProcessManager {
 	 * @return {ProcessWrapper}
 	 */
 	acquire() {
-		let process = this.processes[0];
-		for (let i = 1; i < this.processes.length; i++) {
-			if (this.processes[i].load < process.load) {
-				process = this.processes[i];
-			}
-		}
-		return process;
+		return this.processes.reduce((best, current) => current.load < best.load ? current : best); // Not sure if this is too obvious...?
```

Comment:
Not really obvious. This is clearly `Math.min`, except JavaScript doesn't provide version of that function that takes a callback. You may want to leave the original version here.

Reply:
Hmm, `Array.reduce` exists for precisely this purpose though

Reply:
I'm aware; I'd still prefer a for-loop because it's clearer and probably more performant.

Reply:
Should be fixed @xfix